src/vsr/checkpoint_trailer.zig

```diff
@@ -116,12 +116,7 @@ pub fn CheckpointTrailerType(comptime Storage: type) type {
             checkpoint: *const fn (trailer: *Self) void,
         } = .none,
 
-        comptime {
-            assert(FreeSet.Word == schema.FreeSetNode.Word);
-            assert(chunk_size_max % @sizeOf(FreeSet.Word) == 0);
-        }
-
-        pub fn init(allocator: mem.Allocator, buffer_size: usize) !Self {
+        pub fn init(allocator: mem.Allocator, buffer_size: usize, trailer_type: TrailerType) !Self {
```

Comment:
Femto-nit, but mabye put type before the buffer? As it is the most important argument that
determines the rest.